### DISCUSSION TITLE
State is not sync in x-for

### DISCUSSION THREAD

**Discussion by jonyx225** [alpinejs/alpine#1400] [7. Bugs (archived)]
I have two arrays in my scope
{
 pets: [],
 filteredPets:[]
}

Once I update filteredPets = pets.filter(...)

I get correct filteredPets,

but if I use x-for to print filteredPets, it prints pets, not filteredPets, but the length is filteredPets.length

**Comment by jonyx225** [comment]
I guess DOM is not  updated correctly if I load DOM elements from array,

**Comment by KevinBatdorf** [comment]
Do you have a demo online? It seems to work for me like this: https://codepen.io/KevinBatdorf/pen/MWjqGJp

**Comment by jonyx225** [comment]
Yeah, it works for the first time on rendering.
The thing happens on re-rendering,
On re-rendering previous DOM is not cleared, thus it affects updated DOM,

**Comment by jonyx225** [comment]
https://user-images.githubusercontent.com/47555611/104191996-3b776600-53ec-11eb-8364-6c3b0e268209.mp4

Please check this out
1. filter "All" is rendered for the first time - fine
2. if I select "adopted", it should render "adopted" pet, but it renders 3 first pets (there are 3 adopted pets)
3. if I select "special needs" and then select "adopted", it renders correctly, 3 adopted pets

I guess this is all about DOM re-render when state is updated

**Comment by arceus03** [comment]
@PonyJackal Can you provide a codesandbox link?
